### ISSUE TITLE
Is it possible to choose default auth source?

### ISSUE THREAD

**Issue by andloh** [go-gitea/gitea#7633]
- Gitea version (or commit ref): 1.10.0+dev-60-gc5f9c0418
- Git version:
- Operating system: linux
- Database (use `[x]`):
  - [ x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x ] Not relevant

**Comment by lafriks** [comment]
How do you mean default? They all work at the same time

**Comment by andloh** [comment]
Thanks for answering
More like if its possible to disable some of the auth sources. E.g so that users can just interact with the oidc auth, and not the default gitea one.

**Comment by lunny** [comment]
@andloh Gitea support LDAP/mail/OAuth2 and etc. Is that what you mean?

**Comment by andloh** [comment]
No. The auth support is great. But i would like to disable some of them. So that users only have one auth source to interact with on the login page

**Comment by army1349** [comment]
Once [role mapping](https://github.com/go-gitea/gitea/issues/12493) is possible option to force OAuth2 auth source would be nice.

**Comment by ailox** [comment]
In addition to what @army1349 said, OIDC/OAuth2 Autoregistration (worked on in https://github.com/go-gitea/gitea/pull/5123) would also be useful to make login for existing OIDC accounts seamless.

**Comment by tacerus** [comment]
I support this issue.

**Comment by tacerus** [comment]
I worked around this by adding a redirect to Gitea's oauth URL in my webserver (nginx) in order to directly forward users to our OIDC/SSO logon page upon clicking "Sign in" in Gitea - maybe it helps others:

```
location                /user/login {
return                  302 https://git.com.de/user/oauth2/LibertaCasaSSO;
}

```

The last bit is the name of the authentication provider that was set in Gitea - best to just visit the Gitea sign-in page regularly, right-clicking on the SSO/OIDC button, and selecting the browsers "Copy Link" option.

**Comment by wxiaoguang** [comment]
> Thanks for answering More like if its possible to disable some of the auth sources. E.g so that users can just interact with the oidc auth, and not the default gitea one.

The same request as #13606, will be implemented by #32687
